x-pack/test/cases_api_integration/common/lib/validation.ts

```diff
@@ -29,13 +30,12 @@ export function validateCasesFromAlertIDResponse(
  * Compares two arrays to determine if they are sort of equal. This function returns true if the arrays contain the same
  * elements but the ordering does not matter.
  */
-export function arraysToEqual(array1?: object[], array2?: object[]) {
+export function arraysToEqual<T>(array1?: T[], array2?: T[]) {
   if (!array1 || !array2 || array1.length !== array2.length) {
     return false;
   }
 
-  const array1AsSet = new Set(array1);
-  return array2.every((item) => array1AsSet.has(item));
+  return xorWith(array1, array2, isEqual).length === 0;
```

Comment:
I think `isEqual` is sufficient, correct? According to the docs, it performs a deep comparison between the arrays.

Reply:
We talked about this offline, `isEqual` requires the arrays to have the same order. This function is ignoring order which is why we need `xorWith`.